app/helpers/application_helper.rb

```diff
@@ -7,7 +7,24 @@ def content_tagger_url
     Plek.new.external_url_for('content-tagger')
   end
 
-  def preview_url(slug)
-    "#{Plek.new.external_url_for('draft-origin')}/#{slug.sub(/^\//, '')}"
+  def preview_url(slug, auth_bypass_id: nil)
+    url = "#{Plek.new.external_url_for('draft-origin')}/#{slug.sub(/^\//, '')}"
+
+    if auth_bypass_id.present?
+      token = jwt_token(auth_bypass_id)
+      url << "?token=#{token}"
+    end
+
+    url
+  end
+
+private
+
+  def jwt_token(auth_bypass_id)
+    JWT.encode({ 'sub' => auth_bypass_id }, jwt_auth_secret, 'HS256')
```

Comment:
The `jwt_auth_secret` uses a bit of indirection. Would it be simpler to use the ENV variable here directly?

Reply:
OK, but I meant more that `Rails.application.config.jwt_auth_secret` is a bit of an indirection. It's two layers of environment specific configuration. If you move this into a class I'd consider dropping some.